### PR TITLE
OOP school library: add basic UI

### PR DESCRIPTION
**Pull request by uno36**
### Project requirements
- Watch the video with the UI example again (you have seen it while reading the Sneak Peek).
    - [UI example](https://www.youtube.com/watch?v=vkkgrhD6aXQ)
- Your console app should behave in the same way as it is in the example.
- Create a `app.rb` file that will serve as your console app entry-point. It should have methods that do the following:
  - List all books.
  - List all people.
  - Create a person (teacher or student, not a plain `Person`).
  - Create a book.
  - Create a rental.
  - List all rentals for a given person id.
- In your `main.rb` define the entry point, this will be a method called `main` that is invoked at the end of your file. This method should do the following:
  - Present the user with a list of options to perform.
  - Lets users choose an option.
  - If needed, ask for parameters for the option.
  - Have a way to quit the app.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    puts 'Listing all people:'
+    @people.each { |person| puts "#{person.class}: #{person.name} (ID: #{person.id})" }
+  end
+
+  def create_person
+    puts 'Is the person a student or a teacher? (S/T)'
+    person_type = gets.chomp.upcase
+
+    if person_type == 'S'
+      create_student
```
Comment (OyePriscilla) [inline app.rb]: - [ ] Please ensure to implement the `app.rb` as stated in the reqirement, all these methods (**`from line 24 -123, 158-168`**) supposed to be in `app.rb` file. main.rb should only be defined as the entry point, with a method that is invoked at the end of your file. ℹ️ You can reference this as a [refresher](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/lessons/console_apps.md#console-app-entry-point).

Untitled

File: app.rb [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+  end
+
+  def create_rental
+    puts "Enter the person's ID who is renting the book:"
+    person_id = gets.chomp.to_i
+    person = find_person_by_id(person_id)
+    if person.nil?
+      puts "Person with ID #{person_id} not found."
+      return
+    end
```
Comment (OyePriscilla) [inline app.rb]: - [ ] Please kindly implement the create_rental method as it's stated in the requirement in the following order 👇 

**`Select a book from the list by Author`
`Select a person from list by number (not id)`**
**`Enter Date`**
Kindly watch [this video](https://www.youtube.com/watch?v=vkkgrhD6aXQ) again for confirmation. Thanks. 🙏

**Review by OyePriscilla — CHANGES_REQUESTED** [review]
Hi @uno36 👏 👏, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

so-close

## Highlights
Implemented student's school library UI ✔️ 
`main.rb` define the entry point ✔️ 
`app.rb` file conatins given methods ❎ 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Lizdev-05 — APPROVED** [review]
Hi @uno36 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



## Highlights

:heavy_check_mark: Professional Readme file :+1: 
:heavy_check_mark: Good commit history :+1:
:heavy_check_mark: No linter error :+1: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**